DS13/code/game/objects/replicator.dm

```diff
@@ -0,0 +1,322 @@
+/obj/machinery/replicator //Coded by "AbsurdlyLudicrous", tweaked to use voice + cleaned up by Kmc.
+	name = "replicator"
+	desc = "An advanced energy -> matter synthesizer which is charged by <i>biomatter</i> and power. Click it to see the menu and simply say what you want to it."
+	icon = 'DS13/icons/obj/replicator.dmi'
+	icon_state = "replicator"
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 40
+	pixel_y = 32 //So it glues to the wall
+	anchored = TRUE
+	flags_1 = HEAR_1 //So it can hear you order your food
+	circuit = /obj/item/circuitboard/machine/replicator
+	var/list/menutier1 = list("rice", "egg", "ration pack", "glass") //It starts off terribly so the chef isn't replaced. You can then upgrade it via RnD to give actual food.
+	var/list/menutier2 = list("burger", "steak", "tea earl grey", "coffee", "fries","onion rings", "pancakes")
+	var/list/menutier3 = list("cheese pizza", "meat pizza", "mushroom pizza", "meat pizza", "pineapple pizza", "donkpocket pizza", "vegetable pizza")
+	var/list/menutier4 = list("cake batter", "dough","egg box", "flour", "milk", "enzymes", "cheese wheel", "meat slab","an insult to pizza")
+	var/list/all_menus = list() //All the menu items. Built on init(). We scan for menu items that've been ordered here.
+	var/list/menualtnames = list("nutrients", "donk pizza", "veggie pizza", "surprise me", "you choose", "something", "i dont care","slab of meat","nutritional supplement")
+	var/list/temps = list("cold", "warm", "hot", "extra hot")
+	var/menutype = "ready" //Tracks what stage the machine's at. If it's replicating the UI pops up with "please wait!"
+	var/fuel = 50 //Starts with a bit of fuel for lazy chefs.
+	var/bingrade = 1
+	var/lasergrade = 1
+	var/manipulatorgrade = 1
+	var/scannergrade = 1
+	var/emagged = FALSE
+	var/ready = TRUE //Are we ready to make some food?
```

Comment:
Can you bloody tell me why this is here and ``menutype`` is here? they seem to both be doing the same thing